airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -0,0 +1,445 @@
+    Endpoint: https://api.intercom.io/data_attributes?model=company
+    """
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {"model": "company"}
+
+
+class ContactAttributes(DataAttributes):
+    """Return list of all data attributes belonging to a workspace for contacts.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-data-attributes
+    Endpoint: https://api.intercom.io/data_attributes?model=contact
+    """
+
+    def request_params(
+            self,
+            stream_state: Mapping[str, Any],
+            stream_slice: Mapping[str, any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {"model": "contact"}
+
+
+class Tags(IntercomStream):
+    """Return list of all tags.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-tags-for-an-app
+    Endpoint: https://api.intercom.io/tags
+    """
+
+    primary_key = "name"
+
+    def path(
+            self,
+            stream_state: Mapping[str, Any] = None,
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "tags"
+
+
+class Teams(IntercomStream):
+    """Return list of all teams.
+    API Docs: https://developers.intercom.com/intercom-api-reference/reference#list-teams
+    Endpoint: https://api.intercom.io/teams
+    """
+
+    primary_key = "name"
+    data_fields = ["teams"]
+
+    def path(
+            self,
+            stream_state: Mapping[str, Any] = None,
+            stream_slice: Mapping[str, Any] = None,
+            next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "teams"
+
+
+class SourceIntercom(AbstractSource):
+    """
+    Source Intercom fetch data from messaging platform.
+    """
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        authenticator = TokenAuthenticator(token=config["access_token"])
+        try:
+            url = f"{IntercomStream.url_base}/tags"
+            auth_headers = {"Accept": "application/json",
+                            **authenticator.get_auth_header()}
+            session = requests.get(url, headers=auth_headers)
+            session.raise_for_status()
+            return True, None
+        except requests.exceptions.RequestException as e:
+            return False, e
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        now = date.today()
+
+        start_date = config.get("start_date")
+        if start_date and isinstance(start_date, str):
+            start_date = datetime.strptime(config["start_date"], "%Y-%m-%dT%H:%M:%S%z")
+        config["start_date"] = start_date or now - timedelta(days=365)  # set to 1 year ago by default
```

Comment:
this line is not needed. as far as I see there is no such limit from API side

Reply:
Done